appcues/src/main/java/com/appcues/analytics/AutoPropertyDecorator.kt

```diff
@@ -77,8 +81,19 @@ internal class AutoPropertyDecorator(
             "_lastScreenTitle" to previousScreen,
             "_sessionPageviews" to sessionPageviews,
             "_sessionRandomizer" to sessionRandomId,
+            "_pushPrimerEligible" to getPushPrimerEligible()
         ).filterValues { it != null }.mapValues { it.value as Any }
 
+    private fun getPushPrimerEligible(): Boolean {
+        val activity = AppcuesActivityMonitor.activity ?: return false
+        // VERSION guard because of Manifest.permission.POST_NOTIFICATIONS
+        return if (VERSION.SDK_INT >= VERSION_CODES.TIRAMISU) {
+            ActivityCompat.shouldShowRequestPermissionRationale(activity, Manifest.permission.POST_NOTIFICATIONS)
+        } else {
+            false
+        }
+    }
```

Comment:
probably ok, but wondering if there is a way we could avoid having to check this on every single decorate call? I guess that gets complicated in some cases to know when to refresh it. Maybe this is low overhead and no big deal.

Reply:
open to ideas, I cant think of a way to keep this "in cache" in a reliable way

Reply:
open to ideas, I cant think of a way to keep this "in cache" in a reliable way